Newsletters: add 158 (2021-07-21)

harding opened the pull request:
- [x] Topics @harding 
- [x] Releases/RCs @harding 
- [x] Clients/services @bitschmidty 
- [x] `Bitcoin Core #22112` Force port 0 in I2P @jnewbery 
- [x] `Rust-Lightning #975` Make the base fee configurable in ChannelConfig @dongcarl 

Note: adds an entry to the style guide.  As always, let me know if anyone disagrees and I'll move that to a separate PR (or realize it's dumb and just delete it).

jnewbery commented on STYLE.md:
```diff
@@ -77,6 +77,9 @@ following abbreviations we assume readers will already know.
 #### Forbidden abbreviations
```
Perhaps this should be "Forbidden terms" or similar, since these aren't really abbreviations.

jnewbery commented on _includes/specials/taproot/en/04-why-wait.md:
```diff
@@ -0,0 +1,77 @@
+  before you think it's safe, that's their risk to take.
+
+[^timelocked-trb]:
+    Users who want to receive a P2TR payment in the first taproot block
+    should generate an address they don't share with anyone and then
```
Even with the disclaimer at the end, this sounds a little bit like encouragement. I'd be tempted to drop this footnote.

bitschmidty replied:
Im tempted to keep it, personally.

harding replied:
I don't have a strong opinion, so feel free to drop it without further input from me.  My argument in favor of keeping it is that I suspect that publishing without this information will lead to someone excitedly telling us that there's a nifty trick people can use to eliminate the danger.  In fact, AJ did exactly that when I mentioned I was writing this section (but, unknown to him, I had already written this footnote).

jnewbery replied:
Let's keep it

jnewbery commented on _includes/specials/taproot/en/04-why-wait.md:
```diff
@@ -0,0 +1,77 @@
+In short:
+
+- {{ante_trb}}: last block where anyone can spend money sent to a P2TR-style output
+- {{site.trb}}: first block where P2TR outputs can only be spent according to
+  the [BIP341][] and [BIP342][] rules.
```
```suggestion
- {{site.trb}}: first block where P2TR outputs can only be spent if they satisfy the [BIP341][] and [BIP342][] rules.
```

(otherwise might be interpreted as being equivalent to "according to the BIP341 and BIP342 rules, this is the first block where P2TR outputs can [only] be spent")

jnewbery commented on _includes/specials/taproot/en/04-why-wait.md:
```diff
@@ -0,0 +1,77 @@
+  the [BIP341][] and [BIP342][] rules.
+- {{safe_trb}}: a reasonable block at which wallets can start giving their
+  users [bech32m][topic bech32] receiving addresses for P2TR outputs
+- None of the above changes the advice given in the [first part][taproot
```
Should this be a bullet point? It seems weird to have three bullets that start with a block height, and then this additional note.

jnewbery commented on _posts/en/newsletters/2021-07-21-newsletter.md:
```diff
@@ -0,0 +1,110 @@
+  channels].
+
+- [C-Lightning #4646][] makes two changes in preparation for removing
+  old behavior.  The first change assumes all nodes support the
```
```suggestion
  old behavior.  The first change assumes nodes support the
```

I don't think the 'all' is necessary here, and it jars a bit with the fact that nodes can signal non-support for TLV encoding.

bitschmidty commented on _posts/en/newsletters/2021-07-21-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+- **Sparrow 1.4.3 supports P2TR:**
+  Sparrow's [1.4.3 release][sparrow 1.4.3] supports [single-sig P2TR
+  wallets][taproot series 4] on [signet][topic signet] and regtest. The release
+  also supports [sending to P2TR addresses][taproot series 1].
```
Should P2TR addresses be swapped to bech32m per the style guide change above?

harding replied:
From a quick look at their [diff](https://github.com/sparrowwallet/drongo/commit/967a2c2026f15a6b458d055e33ac488e53df65fd#diff-85d66c4a8decae8670cf725b591dfa8c626a569ed607aa8e887964795a9df9b8), it seems to me that they only added support for bech32m with v1 segwit (P2TR).  So what you wrote is not only correct but precise.  That said, it would be more in line with the style guide suggestion to say something like "sending to bech32m addresses for P2TR".  Your choice.

bitschmidty replied:
Took this suggestion

adamjonas commented on _posts/en/newsletters/2021-07-21-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+  old behavior.  The first change assumes all nodes support the
+  TLV-style encoding added in 2019 (see [Newsletter #55][news55 tlv]).
+  Only nodes that explicitly indicate they don't support TLV encoding
+  will by treated differently.
```
```suggestion
  will be treated differently.
```

adamjonas commented on _posts/en/newsletters/2021-07-21-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+
+- [C-Lightning #4614][] updates the `listchannels` RPC with a new
+  optional `destination` parameter that can be used to only return
+  channels that lead to requested node.
```
```suggestion
  channels that lead to a requested node.
```

Xekyo replied:
```suggestion
  channels that lead to the requested node.
```

harding commented on _posts/en/newsletters/2021-07-21-newsletter.md:
```diff
@@ -0,0 +1,157 @@
+  execution of Discreet Log Contracts (DLCs) on the Bitcoin blockchain using an
+  oracle. The announcement concludes by mentioning they also plan to use [schnorr
+  signatures][topic schnorr signatures] and [Point Time Locked Contracts
+  (PTLCs)][topic ptlc] to implement [DLCs on Lightning][suredbits blog dlcs ln].
```
I'm going to change these last few words to "DLCs compatible with LN".  What they're describing can lead to DLCs being routed over LN, but as the last lines of the blog post you link to indicate, this conflicts with the way capital is deployed on LN.  I'm personally unaware of any solution to this problem, so I'd like to slightly tone down the enthusiasm here.

As always, please feel free to revert if you disagree.

bitschmidty replied:
Yes, good nuance, thanks for the change.

adamjonas commented on _posts/en/newsletters/2021-07-21-newsletter.md:
```diff
@@ -119,7 +119,10 @@ BOLTs][bolts repo].*
-- [Rust-Lightning #975][] Make the base fee configurable in ChannelConfig FIXME:dongcarl
+- [Rust-Lightning #975][] makes the base payment forwarding fee configurable
+  with a default value of 1 satoshi (the market rate as of July 2021).
+  Previously, the base fee was determined pesimally by estimating the fee
```
pessimistically?

harding commented:
Edits made, hopefully addressing all feedback (thanks!).  Also reviewed the other contributor's content and made a few edits, updated releases, and added topic links.

Thanks everyone!

jnewbery commented on _topics/en/anonymity-networks.md:
```diff
@@ -94,6 +94,9 @@ optech_mentions:
   - title: "Bitcoin Core #22050 drops support for deprecated version 2 Tor onion services"
     url: /en/newsletters/2021/06/09/#bitcoin-core-22050
 
+  - title: " Bitcoin Core #22112 changes the assumed port for I2P addresses to be 0 instead of 8333"
```
```suggestion
  - title: "Bitcoin Core #22112 changes the assumed port for I2P addresses to be 0 instead of 8333"
```

(doesn't impact generated site, just triggers jnewbery)

jnewbery reviewed (commented):
All looks good to me. Just two outstanding comments for @bitschmidty to look at before merge:

- https://github.com/bitcoinops/bitcoinops.github.io/pull/608#discussion_r673557611
- https://github.com/bitcoinops/bitcoinops.github.io/pull/608#discussion_r673577649

bitschmidty commented:
Accepted any outstanding suggestions. Last call on review!

bitschmidty commented:
Any we're off! Thanks @harding @jnewbery @dongcarl @Xekyo @adamjonas
